client/src/wayland_obj/manager.rs

```diff
@@ -0,0 +1,168 @@
+};
+
+thread_local! {
+    static WAYLAND: RefCell<Option<WaylandManager>> = RefCell::new(None);
+}
+
+struct WaylandManager {
+    compositor_manager: WlCompositorManager,
+    output_manager: WlOutputManager,
+    seat_manager: WlSeatManager,
+    shell_manager: LayerShellManager,
+    shm_manager: WlShmManager
+}
+
+// pub fn create_surface() -> Result<WlSurface, ()> {
+//     WAYLAND.with(|wayland| {
+//         let wayland = wayland.borrow();
+//         let wayland = wayland.as_ref().expect("Wayland not initialized");
+
+//         wayland.compositor_manager.create_surface()
+//     })
+// }
+
+pub fn create_buffer(size: Size) -> Result<Buffer, ()> {
+    WAYLAND.with(|wayland| {
+        let wayland = wayland.borrow();
+        let wayland = wayland.as_ref().expect("Wayland not initialized");
+
+        wayland.shm_manager.create_buffer(size)
+    })
+}
+
+pub fn create_layer_surface() -> Result<LayerSurface, ()> {
+    WAYLAND.with(|wayland| {
+        use layer_shell::Layer;
+        let wayland = wayland.borrow();
+        let wayland = wayland.as_ref().expect("Wayland not initialized");
+
+        let wl_surface = wayland.compositor_manager.create_surface()?;
+
+        wayland
+            .shell_manager
+            .create_layer_surface(wl_surface, None, Layer::Top)
+    })
+}
+
+pub fn global_callback(event: GlobalEvent, registry: WlRegistry) {
+    WAYLAND.with(|wayland| {
+        let wayland = &mut wayland.borrow_mut();
+        let wayland = wayland.as_mut().expect("Wayland not initialized");
+        match event {
+            GlobalEvent::New {
+                id,
+                interface,
+                version
+            } => wayland.new_global(registry, id, &interface, version),
+            _ => unimplemented!("GlobalEvent")
+        }
+    })
+}
+
+pub fn init_wayland() -> (Display, EventQueue) {
+    let (display, event_queue) = Display::connect_to_env().unwrap_or_else(|err| {
+        use std::{env, process::exit};
+        use wayland_client::ConnectError::*;
+        match err {
+            NoWaylandLib => error!("Could not find Wayland library, is it installed and in PATH?"),
+            NoCompositorListening => {
+                error!("Could not connect to Wayland server. Is it running?");
+                error!(
+                    "WAYLAND_DISPLAY={}",
+                    env::var("WAYLAND_DISPLAY").unwrap_or_default()
+                );
+            },
+            InvalidName => error!("Invalid socket name provided in WAYLAND_SOCKET"),
+            XdgRuntimeDirNotSet => error!("XDG_RUNTIME_DIR must be set"),
+            InvalidFd => error!("Invalid socket provided in WAYLAND_SOCKET")
+        }
+        exit(1);
+    });
+
+    WAYLAND.with(|wayland| {
+        let wayland = &mut wayland.borrow_mut();
+        let way_man = WaylandManager::new(crate::lua::WaylandHandler {});
+        (*wayland).replace(way_man);
+    });
+
+    (display, event_queue)
+}
+
+impl WaylandManager {
```

Comment:
`WaylandManager`? `WaylandFacade`? `WaylandGlobalImplementor`?

Reply:
What it is so far doing is managing the globals advertised by the Wayland compositor, so maybe `WaylandGlobalManager`?